crypto3/libs/blueprint/include/nil/blueprint/zkevm_bbf/input_generators/basic_input_generator.hpp

```diff
@@ -1000,29 +1000,45 @@ namespace nil {
                 }
 
                 virtual void codecopy() override{
-                    std::size_t dst = std::size_t(stack[stack.size() - 1]);
-                    std::size_t src = std::size_t(stack[stack.size() - 2]);
-                    std::size_t length = std::size_t(stack[stack.size() - 3]);
+                    constexpr static const std::size_t max_offset = 65536;
+                    auto dst = stack[stack.size() - 1];
+                    auto src = stack[stack.size() - 2];
+                    auto lgt = stack[stack.size() - 3];
 
                     _zkevm_states.back().load_stack(stack, 3);
                     append_stack_reads(3);
 
                     zkevm_basic_evm::codecopy();
-                    copy_event cpy = codecopy_copy_event(
-                        bytecode_hash,
-                        src,
-                        call_id,
-                        dst,
-                        rw_counter,
-                        length
-                    );
-                    for( std::size_t i = 0; i < length; i++){
-                        _short_rw_operations.push_back(memory_rw_operation(
-                            call_id, dst + i, rw_counter++, true, memory[dst + i]
-                        ));
-                        cpy.push_byte(memory[dst+i]);
+                    if (memory.size() != 0) {
+                        std::size_t distance = std::size_t(dst);
+                        std::size_t length = std::size_t(lgt);
+                        std::size_t source = src < max_offset ? std::size_t(src) : max_offset;
+
+                        //Only copy real bytes
+                        std::size_t real_length = (source >= bytecode.size()) ? 0 : std::min(length, bytecode.size() - source);
+                        copy_event cpy = codecopy_copy_event(
+                            bytecode_hash,
+                            source,
+                            call_id,
+                            distance,
+                            rw_counter,
+                            real_length
+                        );
+                        //TODO?
+                        //Do we need a second copy event for the fake copy?
```

Comment:
1. Add this copy event only if real_length != 0.
2. Do we check EVM's behaviour? If 0 is written to the remaining memory, an additional copy event is needed